docs/docs/higher-order-components.md

```diff
@@ -0,0 +1,488 @@
+- Inside the listener, call `setState` whenever the data source changes.
+- On unmount, remove the change listener.
+
+You can imagine that in a large app, this same pattern of subscribing to `DataSource` and calling `setState` will occur over and over again. We want an abstraction that allows us to define this logic in a single place and share them across many components. This is where HOCs excel.
+
+We'll create a function called `withSubscription` that works like this:
+
+```js
+const CommentListWithSubscription = withSubscription(
+  // The first parameter is the wrapped component
+  CommentList,
+  // The second parameter retrieves the data we're interested in
+  (DataSource) => DataSource.getComments()
+);
+
+const BlogPostWithSubscription = withSubscription(
+  CommentList,
+  // An optional second argument passes the current props
+  (DataSource, props) => DataSource.getBlogPost(props.id)
+});
+```
+
+And here's what the implementation of `withSubscription` looks like:
+
+```js
+// This function takes a component...
+function withSubscription(WrappedComponent, selectData) {
+  // ...and returns another component...
+  return class extends React.Component {
+    constructor(props) {
+      super();
+      this.handleChange = this.handleChange.bind(this);
+      this.state = {
+        data: selectData(DataSource, props)
+      };
+    }
+
+    componentDidMount() {
+      // ... that takes care of the subscription...
+      DataSource.addChangeListener(this.handleChange);
+    }
+
+    componentWillUnmount() {
+      DataSource.removeChangeListener(this.handleChange);
+    }
+
+    handleChange() {
+      this.setState({
+        comments: selectData(DataSource, this.props)
+      });
+    }
+
+    render: function() {
+      // ... and renders the wrapped component with the fresh data!
+      // Notice that we pass through any additional props
+      return <WrappedComponent data={this.state.data} {...this.props} />;
+    }
+  });
+}
+```
+
+And that's it! The wrapped component receives a single prop, `data`, which it uses to render its output. The HOC isn't concerned with how or why the data is used, and the wrapped component isn't concerned with where the data came from.
+
+Because `withSubscription` is a normal function, you can add as many or as few arguments as you like. For example, you may want to make the name of the `data` prop configurable, to further isolate the HOC from the wrapped component. Or you could accept an argument that configures `shouldComponentUpdate`, or one that configures the data source. These are all possible because the HOC has full control over how the component is defined.
+
+Note that like components, the contract between `withDataSubscription` and the wrapped component is entirely props-based. This makes it easy to swap one HOC for a different one, as long as they provide the same props to the wrapped component. An example of how this may be useful is if you change data-fetching libraries.
```

Comment:
An example of how this may be useful is if you change data-fetching libraries.

=>

This may be useful if you change data-fetching libraries.

Less is more :-)